### ISSUE TITLE
Proxy without portforwarding

### ISSUE THREAD

**Issue by PaulDotSH** [dizda/fast-socks5#28]
Hi, is there any plan to implement starting a proxy server using something like "Long Polling" to enable starting proxy servers without port forwarding?

Alternatives to long polling might also be
Short Polling (wastes more cpu and network resources)
Websockets (wastes way more resources, and there is no auto reconnection, and no auth system)
Server-Sent Events (harder to implement and wastes more resources)

Reason: some people like myself have port forwarding blocked by isps and would like to still be able to use their home network as a proxy

**Comment by dizda** [comment]
I'm not sure to understand correctly.
Long polling usually refers to HTTP protocol, but this library is a socket protocol, which means that it could handle HTTP, but also any other ports.

**Comment by PaulDotSH** [comment]
> I'm not sure to understand correctly. Long polling usually refers to HTTP protocol, but this library is a socket protocol, which means that it could handle HTTP, but also any other ports.

Yes, it refers to http usually, but could be implemented for socks to enable hosting a proxy without the need of port forwarding

**Comment by dizda** [comment]
Sorry, this is out of scope
